File: problem14/Ruby/problem14.rb
def getCount(num)
	length = 1
	while (num != 1)
		num = num % 2 == 0 ? num / 2 : 3 * num + 1
		length += 1
	end
	return length
end

def main()
	key = 0
	value = 0
	1.upto(1000000) do |i|
		length = getCount(i)
		key,value = i,length if getCount(i) > value
	end
	puts key,value
end

main()

File: problem14/C/problem14.c
/*
compile: GCC 4.8.2
author: firebroo
email: bianliqiang12@gmail.com
*/

#include<stdio.h>

int getCount(long num) {
	long length = 1;
	while (num != 1) {
		if(num % 2 == 0) {
			num = num / 2;
		}else {
			num = num * 3 + 1;
		}
		length += 1;
	}
	return length;
}

int main(void) {
	long key = 0;
	long value = 0;
	long length = 0;
	for(long i=1;i<1000000;i++) {
		length = getCount(i);
		if(length > value) {
			key = i;
			value = length;
		}
	}
	printf("%ld\t%ld\n", key, length);
	return 0;
}
